click/core.py

```diff
@@ -329,9 +329,9 @@ def __init__(self, command, parent=None, info_name=None, obj=None,
                and parent.auto_envvar_prefix is not None and \
                self.info_name is not None:
                 auto_envvar_prefix = '%s_%s' % (parent.auto_envvar_prefix,
```

Comment:
You're using the prefix here as well, but don't replace dashes in there (even though putting dashes in the prefix would be a a case of "garbage in, garbage out").

Maybe it'd be better to just do the replacement in line 335 since this would cover everything?

Reply:
Your reasoning applies to the `.upper()` call then right.
I just assumed `parent.auto_envvar_prefix` would have had the `.replace('-', '_')` treatment already.

Reply:
@ThiefMaster can we keep the discussion going please?